spring-boot-examples/workflows/src/main/java/io/dapr/springboot/examples/wfp/WorkflowPatternsRestController.java

```diff
@@ -0,0 +1,140 @@
+limitations under the License.
+*/
+
+package io.dapr.springboot.examples.wfp;
+
+import io.dapr.spring.workflows.config.EnableDaprWorkflows;
+import io.dapr.springboot.examples.wfp.chain.ChainWorkflow;
+import io.dapr.springboot.examples.wfp.child.ParentWorkflow;
+import io.dapr.springboot.examples.wfp.continueasnew.CleanUpLog;
+import io.dapr.springboot.examples.wfp.continueasnew.ContinueAsNewWorkflow;
+import io.dapr.springboot.examples.wfp.externalevent.Decision;
+import io.dapr.springboot.examples.wfp.externalevent.ExternalEventWorkflow;
+import io.dapr.springboot.examples.wfp.fanoutin.FanOutInWorkflow;
+import io.dapr.springboot.examples.wfp.fanoutin.Result;
+import io.dapr.workflows.client.DaprWorkflowClient;
+import io.dapr.workflows.client.WorkflowInstanceStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+@RestController
+@EnableDaprWorkflows
+public class WorkflowPatternsRestController {
+
+  private final Logger logger = LoggerFactory.getLogger(WorkflowPatternsRestController.class);
+
+  @Autowired
+  private DaprWorkflowClient daprWorkflowClient;
+
+  private Map<String, String> ordersToApprove = new HashMap<>();
+
+  @Bean
+  public CleanUpLog cleanUpLog(){
+    return new CleanUpLog();
+  }
+
+  /**
+   * Run Chain Demo Workflow
+   * @return the output of the ChainWorkflow execution
+   */
+  @PostMapping("wfp/chain")
+  public String chain() throws TimeoutException {
+    String instanceId = daprWorkflowClient.scheduleNewWorkflow(ChainWorkflow.class);
+    logger.info("Workflow instance " + instanceId + " started");
+    return daprWorkflowClient
+            .waitForInstanceCompletion(instanceId, Duration.ofSeconds(2), true)
+            .readOutputAs(String.class);
+  }
+
+
+  /**
+   * Run Child Demo Workflow
+   * @return confirmation that the workflow instance was created for the workflow pattern child
+   */
+  @PostMapping("wfp/child")
+  public String child() throws TimeoutException {
+    String instanceId = daprWorkflowClient.scheduleNewWorkflow(ParentWorkflow.class);
+    logger.info("Workflow instance " + instanceId + " started");
+    return daprWorkflowClient
+            .waitForInstanceCompletion(instanceId, Duration.ofSeconds(2), true)
+            .readOutputAs(String.class);
+  }
+
+
+  /**
+   * Run Fan Out/in Demo Workflow
+   * @return confirmation that the workflow instance was created for the workflow pattern faninout
+   */
+  @PostMapping("wfp/fanoutin")
```

Comment:
Can you pull out the different patterns here to their respective folder? It will be cleaner to users to have 1 clear example per pattern rather than a shared file used by all examples.

Reply:
The examples are all in different folders, this is just the access point. I can create separate applications per pattern (as it is done in the Dapr University) but that sounds like an overkill for the samples app